src/main/java/org/mifos/connector/mpesa/camel/routes/PaybillRoute.java

```diff
@@ -150,8 +151,9 @@ public void configure() {
                     e.setProperty("CONFIRMATION_REQUEST", obj.toString());
                     //Getting mpesa and workflow transaction id and removing key
                     String mpesaTransactionId = paybillConfirmationRequestDTO.getTransactionID();
-                    String transactionId = mpesaTxnIdStore.get(mpesaTransactionId);
-                    mpesaTxnIdStore.remove(transactionId);
+                    String transactionId = workflowInstanceStore.get(mpesaTransactionId);
+                    workflowInstanceStore.remove(mpesaTransactionId);
+                    reconciledStore.remove(mpesaTransactionId);
```

Comment:
this  reconciledStore.remove to be done as soon as it is read. otherwise the transactions that are not confirmed will be stuck in this Hashmap. 

also we need a way to remove 'workflowInstanceStore' ids for the transactions for which the confirmation is not called. is there a way to do it?